notifications_utils/strftime_codes.py

```diff
@@ -0,0 +1,71 @@
+"""This module provides platform compatible format codes for strftime.
+
+   By default, Python does not check the format codes sent to strftime:
+   these are sent directly to the platform's implementation. This leads
+   developers to use platform specific format codes for strftime that
+   can't easily run on other platforms that are popular and normally
+   easy to support for by the Python language.
+
+   Hence we have this module that makes the translation between platform
+   specific format codes. To refer to the different platform specific
+   codes and what is deemed safe (i.e. ANSI C/ POSIX compatible):
+
+   For windows, see:
+   https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/strftime-wcsftime-strftime-l-wcsftime-l?redirectedfrom=MSDN&view=msvc-160
+
+   For *nix compatible (non-POSIX compatible), see:
+   https://man7.org/linux/man-pages/man3/strftime.3.html
+
+   'somewhat POSIX'/ ANSI C compatible, see
+   https://en.cppreference.com/w/c/chrono/strftime
+"""
+
+import platform
+
+
+SYSTEM = platform.system().lower()
+IS_WINDOWS: bool = SYSTEM == 'windows'
+IS_LINUX: bool = SYSTEM == 'linux'
+IS_MAC: bool = SYSTEM == 'darwin'
+
+NO_PAD_POSIX_CHAR = '-'
+NO_PAD_WINDOWS_CHAR = '#'
+
+NO_PAD_CODES: list = ['d', 'D', 'e', 'F', 'H', 'I', 'j', 'm', 'M',
+                      'r', 'R', 'S', 'T', 'U', 'V', 'W', 'y', 'Y']
+
+
+def _get_no_pad_char() -> str:
+    return NO_PAD_WINDOWS_CHAR if IS_WINDOWS else NO_PAD_POSIX_CHAR
+
+
+def _get_no_pad_code(char: str) -> str:
+    if char not in NO_PAD_CODES:
+        raise ValueError(f'The {char} character is not supported for no padding: {NO_PAD_CODES}')
+    no_pad = _get_no_pad_char()
+    return f'%{no_pad}{char}'
+
+
+def no_pad_code(code: str) -> str:
+    """Gets the non padded format for the given code (i.e. leading zero is removed)"""
+    return _get_no_pad_code(code)
```

Comment:
Can move the other function code here as you're directly calling the private method

Reply:
Just moved it..